internal/scheduler/preempting_queue_scheduler.go

```diff
@@ -193,30 +194,29 @@ func (sch *PreemptingQueueScheduler) Schedule(ctx *armadacontext.Context) (*Sche
 	scheduledAndEvictedJobsById := armadamaps.FilterKeys(
 		scheduledJobsById,
 		func(jobId string) bool {
-			_, ok := evictorResult.EvictedJobsById[jobId]
+			_, ok := evictorResult.EvictedJctxsByJobId[jobId]
 			return ok
 		},
 	)
-	for jobId, job := range evictorResult.EvictedJobsById {
+	for jobId, jctx := range evictorResult.EvictedJctxsByJobId {
 		if _, ok := scheduledJobsById[jobId]; ok {
 			delete(scheduledJobsById, jobId)
 		} else {
-			preemptedJobsById[jobId] = job
+			preemptedJobsById[jobId] = jctx.Job
 		}
 	}
 	maps.Copy(sch.nodeIdByJobId, evictorResult.NodeIdByJobId)
 
 	// Re-schedule evicted jobs/schedule new jobs.
 	// Only necessary if a non-zero number of jobs were evicted.
-	if len(evictorResult.EvictedJobsById) > 0 {
+	if len(evictorResult.EvictedJctxsByJobId) > 0 {
 		// Since no new jobs are considered in this round, the scheduling key check brings no benefit.
 		sch.SkipUnsuccessfulSchedulingKeyCheck()
 		schedulerResult, err = sch.schedule(
 			armadacontext.WithLogField(ctx, "stage", "schedule after oversubscribed eviction"),
 			inMemoryJobRepo,
-			// Only evicted jobs should be scheduled in this round,
-			// so we provide an empty repo for queued jobs.
-			NewInMemoryJobRepository(),
+			// Only evicted jobs should be scheduled in this round.
+			nil,
```

Comment:
I prefer never passing in `nil` here (so that we don't have to to defensively check for `nil` in `schedule`).